torch/csrc/api/include/torch/optim/adam.h

```diff
@@ -53,9 +53,9 @@ class TORCH_API Adam : public Optimizer {
      TORCH_CHECK(defaults.lr() >= 0, "Invalid learning rate: ", defaults.lr());
      TORCH_CHECK(defaults.eps() >= 0, "Invalid epsilon value: ", defaults.eps());
      auto betas = defaults.betas();
-     TORCH_CHECK(std::get<0>(betas) >= 0, "Invalid learning rate: ", std::get<0>(betas));
-     TORCH_CHECK(std::get<1>(betas) >= 0, "Invalid learning rate: ", std::get<1>(betas));
-     TORCH_CHECK(defaults.weight_decay() >= 0, "Invalid learning rate: ", defaults.weight_decay());
+     TORCH_CHECK(std::get<0>(betas) >= 0, "Invalid beta for running average of gradient: ", std::get<0>(betas));
+     TORCH_CHECK(std::get<1>(betas) >= 0, "Invalid beta for running average of squared gradient: ", std::get<1>(betas));
```

Comment:
For the purpose of C++/Python parity, we might want to either have the C++ error message be `Invalid beta parameter at index 0/1` (to match Python version: https://github.com/pytorch/pytorch/blob/8afa001d898914a48d6b9e3d944a99607d2819c1/torch/optim/adam.py#L36-L39), or change the Python error message to match what we have here :)

Reply:
On another note, it might be a good idea to also fix the beta range check to be `0 <= std::get<0>(betas) && std::get<0>(betas) < 1.0`, to make it more consistent with Python version. Thanks a lot for your help!